main.go

```diff
@@ -558,14 +577,19 @@ func (context *Context) serveStatus() error {
 		return err
 	}
 
-	cert, _ := context.cert.GetCertificate(nil)
-	if network != "unix" && cert != nil {
+	if network != "unix" && context.tlsConfigSource.CanServe() {
 		config, err := buildConfig(*enabledCipherSuites)
 		if err != nil {
 			return err
 		}
 		config.ClientAuth = tls.NoClientCert
-		listener = certloader.NewListener(listener, context.cert, config)
+
+		serverConfig, ok := context.tlsConfigSource.GetServerConfig(config)
+		if !ok {
+			logger.Print("TLS config not appropriate for server role")
+			return errors.New("TLS config not appropriate for server role")
```

Comment:
Same question as above, wondering if we could collapse this error-checking logic by adding `MustGetServerConfig()` to `TLSConfigSource` that panics if it's not a server config.

Reply:
This was defensive. I don't imagine this ever happening considering the `CanServe` check above. It felt better to handle the condition than through away the `ok` bool. If you'd rather panic, that's fine, but i'd argue for a helper function that takes a `TLSConfigSource` and calls `GetServerConfig()` v.s. bloating the `TLSConfigSource` interface.